pkg/scalers/activemq_scaler.go

```diff
@@ -0,0 +1,268 @@
+	} else {
+		meta.targetQueueSize = defaultTargetQueueSize
+	}
+
+	if val, ok := config.AuthParams["username"]; ok && val != "" {
+		meta.username = val
+	} else if val, ok := config.TriggerMetadata["username"]; ok && val != "" {
+		username := val
+
+		if val, ok := config.ResolvedEnv[username]; ok && val != "" {
+			meta.username = val
+		} else {
+			meta.username = username
+		}
+	}
+
+	if meta.username == "" {
+		return nil, fmt.Errorf("username cannot be empty")
+	}
+
+	if val, ok := config.AuthParams["password"]; ok && val != "" {
+		meta.password = val
+	} else if val, ok := config.TriggerMetadata["password"]; ok && val != "" {
+		password := val
+
+		if val, ok := config.ResolvedEnv[password]; ok && val != "" {
+			meta.password = val
+		} else {
+			meta.password = password
+		}
+	}
+
+	if meta.password == "" {
+		return nil, fmt.Errorf("password cannot be empty")
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("activemq-%s", val))
+	} else {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("activemq-%s", meta.destinationName))
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+
+	return &meta, nil
+}
+
+func (s *activeMQScaler) IsActive(ctx context.Context) (bool, error) {
+	queueSize, err := s.getQueueMessageCount(ctx)
+	if err != nil {
+		activeMQLog.Error(err, "Unable to access activeMQ management endpoint", "managementEndpoint", s.metadata.managementEndpoint)
+		return false, err
+	}
+
+	return queueSize > 0, nil
+}
+
+// getRestAPIParameters parse restAPITemplate to provide managementEndpoint, brokerName, destinationName
+func getRestAPIParameters(meta activeMQMetadata) (activeMQMetadata, error) {
+	u, err := url.ParseRequestURI(meta.restAPITemplate)
+	if err != nil {
+		return meta, fmt.Errorf("unable to parse ActiveMQ restAPITemplate: %s", err)
+	}
+
+	meta.managementEndpoint = u.Host
+	splitURL := strings.Split(strings.Split(u.Path, ":")[1], "/")[0] // This returns : type=Broker,brokerName=<<brokerName>>,destinationType=Queue,destinationName=<<destinationName>>
+	replacer := strings.NewReplacer(",", "&")
+	v, err := url.ParseQuery(replacer.Replace(splitURL)) // This returns a map with key: string types and element type [] string. : map[brokerName:[<<brokerName>>] destinationName:[<<destinationName>>] destinationType:[Queue] type:[Broker]]
+	if err != nil {
+		return meta, fmt.Errorf("unable to parse ActiveMQ restAPITemplate: %s", err)
+	}
+
+	if len(v["destinationName"][0]) == 0 {
+		return meta, errors.New("no destinationName is given")
+	}
+	meta.destinationName = v["destinationName"][0]
+
+	if len(v["brokerName"][0]) == 0 {
+		return meta, fmt.Errorf("no brokerName given: %s", meta.restAPITemplate)
+	}
+	meta.brokerName = v["brokerName"][0]
+
+	return meta, nil
+}
+
+func (s *activeMQScaler) getMonitoringEndpoint() string {
+	replacer := strings.NewReplacer("<<managementEndpoint>>", s.metadata.managementEndpoint,
```

Comment:
@melisatanrverdi I recall discussing this, but don't remember the outcome: is it possible to turn `s.metadata.restAPITemplate` into a [text/template](https://pkg.go.dev/text/template?utm_source=godoc) and use that to populate these values?